src/viur/core/modules/user.py

```diff
@@ -1312,7 +1312,7 @@ def secondFactorProviderByClass(self, cls) -> UserSecondFactorAuthentication:
     def getCurrentUser(self):
         session = current.session.get()
 
-        if session and session.loaded and (user := session.get("user")):
+        if session and (user := session.get("user")):
```

Comment:
```suggestion
        req = current.request.get()
        if session and (session.loaded or req.is_deferred) and (user := session.get("user")):
```

Reply:
We can do it that way if you like.

But that doesn't really make any difference. If the session is not `loaded`, then `session.get("user")` is `None` anyway and the condition is falsy.

Reply:
Now it works in your case i think. If the session is not loaded we can get the current user if the request is deferred.

Reply:
no, I mean in the non-deferred case it wouldn't matter whether you have
```py
if session and (user := session.get("user")):
```
or
```py
if session and (session.loaded or req.is_deferred) and (user := session.get("user")):
``` 
because if the session is not loaded (`session.loaded` is then `False`) there is no content in the unloaded session object and the `session.get()` cannot load anything and returns the default value `None`.
This means that both statements are logically the same.